### ISSUE TITLE
TTY mode 

### ISSUE THREAD

**Issue by ity-consulting** [lorisleiva/laravel-deployer#136]
Hi, 

I'm getting this error when executing the command "php artisan deploy -o git_tty=true"

`Output:
================
✈︎ Deploying deploy on *********@************* with basic strategy
➤ Executing task deploy:prepare
✔ Ok
➤ Executing task deploy:lock
✔ Ok
➤ Executing task deploy:release
✔ Ok
➤ Executing task deploy:update_code
➤ Executing task deploy:failed
✔ Ok
➤ Executing task deploy:unlock
✔ Ok

Error Output:
================

In Process.php line 1059:

  **TTY mode is not supported on Windows platform.**

deploy [-p|--parallel] [-l|--limit LIMIT] [--no-hooks] [--log LOG] [--roles ROLES] [--hosts HOSTS] [-o|--option OPTION] [--] [<stage>]

  at C:\xampp\htdocs\projects\projetFred\vendor\symfony\process\Process.php:256
    252|      */
    253|     public function mustRun(callable $callback = null, array $env = [])
    254|     {
    255|         if (0 !== $this->run($callback, $env)) {
  > 256|             throw new ProcessFailedException($this);
    257|         }
    258|
    259|         return $this;
    260|     }

  Exception trace:

  1   Symfony\Component\Process\Process::mustRun(Object(Closure))
      C:\xampp\htdocs\projects\projetFred\vendor\lorisleiva\laravel-deployer\src\LaravelDeployer\Commands\BaseCommand.php:110

  2   Lorisleiva\LaravelDeployer\Commands\BaseCommand::process("vendor\bin\dep --file=vendor\lorisleiva\laravel-deployer\.build\deploy.p
hp deploy --option="git_tty=true"")
      C:\xampp\htdocs\projects\projetFred\vendor\lorisleiva\laravel-deployer\src\LaravelDeployer\Commands\BaseCommand.php:58

  Please use the argument -v to see more details.
`

**This is my config file:**

`<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default deployment strategy
    |--------------------------------------------------------------------------
    |
    | This option defines which deployment strategy to use by default on all
    | of your hosts. Laravel Deployer provides some strategies out-of-box
    | for you to choose from explained in detail in the documentation.
    |
    | Supported: 'basic', 'firstdeploy', 'local', 'pull'.
    |
    */

    'default' => 'basic',

    /*
    |--------------------------------------------------------------------------
    | Custom deployment strategies
    |--------------------------------------------------------------------------
    |
    | Here, you can easily set up new custom strategies as a list of tasks.
    | Any key of this array are supported in the `default` option above.
    | Any key matching Laravel Deployer's strategies overrides them.
    |
    */

    'strategies' => [
        //
    ],

    /*
    |--------------------------------------------------------------------------
    | Hooks
    |--------------------------------------------------------------------------
    |
    | Hooks let you customize your deployments conveniently by pushing tasks 
    | into strategic places of your deployment flow. Each of the official
    | strategies invoke hooks in different ways to implement their logic.
    |
    */

    'hooks' => [
        // Right before we start deploying.
        'start' => [
            //
        ],
        
        // Code and composer vendors are ready but nothing is built.
        'build' => [
            'npm:install',
            'npm:production',
        ],
        
        // Deployment is done but not live yet (before symlink)
        'ready' => [
            'artisan:storage:link',
            'artisan:view:clear',
            'artisan:cache:clear',
            'artisan:config:cache',
        ],
        
        // Deployment is done and live
        'done' => [
            'fpm:reload',
        ],
        
        // Deployment succeeded.
        'success' => [
            //
        ],
        
        // Deployment failed.
        'fail' => [
            //
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Deployment options
    |--------------------------------------------------------------------------
    |
    | Options follow a simple key/value structure and are used within tasks
    | to make them more configurable and reusable. You can use options to
    | configure existing tasks or to use within your own custom tasks.
    |
    */

    'options' => [
        'application' => env('APP_NAME', 'Laravel'),
        'repository' => 'git@github.******/******.git',
        'php_fpm_service' => 'php7.2-fpm',
    ],

    /*
    |--------------------------------------------------------------------------
    | Hosts
    |--------------------------------------------------------------------------
    |
    | Here, you can define any domain or subdomain you want to deploy to.
    | You can provide them with roles and stages to filter them during
    | deployment. Read more about how to configure them in the docs.
    |
    */
	
	'hosts' => [
		'my_host' => [
			'deploy_path' => '/home/******/******',
			'http_user' => '*******',
			'multiplexing' => true,
			'sshOptions'   => [
				'UserKnownHostsFile' => '/dev/null',
				'StrictHostKeyChecking' => 'no',
				// ...
			],
		],
	],

    /*
    |--------------------------------------------------------------------------
    | Localhost
    |--------------------------------------------------------------------------
    |
    | This localhost option give you the ability to deploy directly on your
    | local machine, without needing any SSH connection. You can use the
    | same configurations used by hosts to configure your localhost.
    |
    */

    'localhost' => [
        //
    ],

    /*
    |--------------------------------------------------------------------------
    | Include additional Deployer recipes
    |--------------------------------------------------------------------------
    |
    | Here, you can add any third party recipes to provide additional tasks, 
    | options and strategies. Therefore, it also allows you to create and
    | include your own recipes to define more complex deployment flows.
    |
    */

    'include' => [
        //
    ],

    /*
    |--------------------------------------------------------------------------
    | Use a custom Deployer file
    |--------------------------------------------------------------------------
    |
    | If you know what you are doing and want to take complete control over
    | Deployer's file, you can provide its path here. Note that, without
    | this configuration file, the root's deployer file will be used.
    |
    */

    'custom_deployer_file' => false,

];`

It's my first deployment so im using the git_tty option. 

Any thoughts why Im getting this error? 

Thanks!

**Comment by lorisleiva** [comment]
Hi there 👋 

The error is pretty well stated: "**TTY mode is not supported on Windows platform.**"

You don't really need the TTY mode if you have already done a git clone from your host. So you could always do that manually then delete the clone and run the command again without the TTY option.
